waku/node/v2/wakunode2.nim

```diff
@@ -197,14 +191,10 @@ proc query*(w: WakuNode, query: HistoryQuery): HistoryResponse =
   ##
   ## Status: Not yet implemented.
   ## TODO Implement as wrapper around `waku_store` and send RPC.
-  result.messages = newSeq[Message]()
-
-  for msg in w.messages:
-    if msg[0] notin query.topics:
-      continue
-
-    # XXX Unclear how this should be hooked up, Message or WakuMessage?
-    # result.messages.insert(msg[1])
+  # This probably shouldn't be calling the function it currently is.
+  # instead it should be calling with a connection so we call a specific node.
```

Comment:
Yeah, so we need to keep track of one (or possibly more) nodes that support the store protocol. Then dial them.

For MVP, I'd have a single store node stored in state, then query should dial it with HistoryRequest. If no store node exist, it should error.